### PR TITLE
Fix bug where reminder date can be set to a date that has already passed

### PR DESCRIPTION
**Pull request by cashewnade**
The current implementation of Reminder allows a date to be set in the past, which should not be the case. Let's fix the bug to close AY2122S2-CS2103T-T17-3#141

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -226,9 +226,17 @@ public static ReminderDescription parseReminderDescription(String description) t
...
         String trimmedDate = reminderDate.trim();
+
         if (!ReminderDate.isValidDate(trimmedDate)) {
             throw new ParseException(DocumentedDate.MESSAGE_CONSTRAINTS);
         }
-        return ReminderDate.parse(trimmedDate);
+
+        ReminderDate parsedDate = ReminderDate.parse(reminderDate);
+
+        if (parsedDate.getDaysPassed() > 0) {
```
Comment (bloodofme) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:236]: would this throw and error if the user wants to set a reminder of the day itself ?

Comment (bloodofme) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:236]: also for this check, i think its possible to put it in reminderdate class. it should be more slap. you can look at how isValidContactedDate implemented in RecentDate class

**Review by bloodofme — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (pyk595) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:236]: Setting reminders on the day itself is okay, as for the check for is valid contacted date, we refactored it to be in parserUtil also

**Review by pyk595 — APPROVED** [review]

**Review by bloodofme — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ngjunkang) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:240]: if you are doing this, maybe make a method in `Reminder` to not violate LoD:
```java
public boolean isPast() {
    return reminderDate.getDaysPassed() > 0;
}
```

**Review by pyk595 — APPROVED** [review]
LGTM

**Pull request merged**